File: Array/2053. Kth Distinct String in an Array/KthDistinct.java
import java.util.HashMap;

class Solution {
    public String kthDistinct(String[] arr, int k) {
        HashMap<String, Integer> dict = new HashMap<>();

        // 统计每个字符串的出现次数
        for (String str : arr) {
            dict.put(str, dict.getOrDefault(str, 0) + 1);
        }

        int count = 0;
        // 找到第k个独特的字符串
        for (String str : arr) {
            if (dict.get(str) == 1) {
                count++;
                if (count == k) {
                    return str;
                }
            }
        }
        return "";
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        String[] arr = {"a", "b", "a", "c", "b"};
        int k = 2;
        System.out.println(solution.kthDistinct(arr, k)); // 输出 "c"
    }
}

File: Array/2053. Kth Distinct String in an Array/KthDistinct.py
from typing import List
from collections import Counter

class Solution:
    def kthDistinct(self, arr: List[str], k: int) -> str:
        # 使用Counter来统计每个字符串的出现次数
        count = Counter(arr)
        
        # 遍历数组，找到第k个独特的字符串
        for str in arr:
            if count[str] == 1:
                k -= 1
                if k == 0:
                    return str
        
        return ""

# 示例用法
solution = Solution()
arr = ["a", "b", "a", "c", "b"]
k = 2
print(solution.kthDistinct(arr, k))  # 输出 "c"